cheatsheets/gleam-for-elm-users.md

```diff
@@ -0,0 +1,699 @@
+---
+layout: page
+title: Gleam for Elm users
+---
+
+- [Overview](#overview)
+- [Comments](#comments)
+- [Variables](#variables)
+  - [Variables type annotations](#variables-type-annotations)
+- [Constants](#constants)
+- [Functions](#functions)
+  - [Function type annotations](#function-type-annotations)
+  - [Function heads](#function-heads)
+  - [Function overloading](#function-overloading)
+  - [Referencing functions](#referencing-function)
+  - [Calling anonymous functions](#calling-anonymous-functions)
+  - [Labelled arguments](#labelled-arguments)
+- [Modules](#modules)
+  - [Exports](#exports)
+- [Blocks](#blocks)
+- [Data types](#data-types)
+  - [Numbers](#numbers)
+  - [Strings](#strings)
+  - [Tuples](#tuples)
+  - [Records](#records) TODO
+  - [Lists](#lists)
+  - [Dicts](#dicts)
+- [Operators](#operators)
+- [Type aliases](#type-aliases)
+- [Custom types](#custom-types) TODO
+  - [Maybe](#maybe)
+  - [Result](#result)
+- [Commands](#commands) TODO
+- [Architecture](#architecture) TODO
+- [Talking to other languages](#talking-to-other-languages) TODO
+- [Package management](#package-management)
+- [Implementation](#implementation)
+- [Other concepts](#other-concepts)
+  - [Atoms](#atoms)
+
+## Overview
+
+Elm and Gleam have similar goals of providing a robust and sound type system with an approachable
+set of features. Gleam started off with a similar syntax to Elm but as moved to use one that is more
+familiar to many developers.
+
+Where Elm compiles to Javascript, Gleam initially aimed to compile to Erlang. Where Elm is best
+suited for front-end browser based applications, Gleam initially targets a back-end ecosystem.
+
+There is work in progess to compile Gleam to Javascript which opens up the possibility for front-end
+development.
+
+One area in which Elm and Gleam differ in their approach is where Elm does not provide easy access
+to defining foreign-function-interfaces for interacting with Javascript libraries. In contrast to
+this, Gleam makes it easy to define this inferfaces for using Erlang libraries.
+
+## Comments
+
+#### Elm
+
+In Elm, single line comments are written with a `--` prefix and multiline comments are written with `{- -}` and `{-| -}` for documentation comments.
+
+```elm
+-- Hello, Joe!
+
+{- Hello, Joe! 
+   This is a multiline comment.
+-}
+
+{-| Determine the length of a list.
+    length [1,2,3] == 3
+-}
+length : List a -> Int
+length xs =
+  foldl (\_ i -> i + 1) 0 xs
+```
+
+#### Gleam
+
+In Gleam comments are written with a `//` prefix.
+
+```rust
+// Hello, Joe!
+```
+
+Comments starting with `///` are used to document the following statement. Comments starting with `////` are used to document the current module.
```

Comment:
```suggestion
Comments starting with `///` are used to document the following function, constant, or type definition. Comments starting with `////` are used to document the current module.
```

There was some confusion around the word "statement" in discord the other day. I am hoping this language is clearer.